splunk-aws-automation/lambda_code/splunk_cwe_firehose_GD_detective_processor_v0.1/lambda_function.py

```diff
@@ -0,0 +1,360 @@
+        if "instanceDetails" in data['detail']['resource']:
+            data['detail']['detectiveUrls']['ec2Instance']  = prefix_url+data['region']+'#entities/Ec2Instance/'+data['detail']['resource']['instanceDetails']['instanceId']+'?scopeStart='+scopeStart+'&scopeEnd='+scopeEnd
+        
+        ## Get nework intefaces info to form URLs
+        ## Get private IP addresses from network interfaces
+        if "networkInterfaces" in data['detail']['resource']['instanceDetails']:            
+            privateIPs = find_key_value_pairs(data['detail'], 'networkInterfaces')
+            
+            index = 0
+            if len(privateIPs) > 0:
+                for k, v in privateIPs:
+                    if (len(privateIPs) == 1 and len(v) == 1):
+                        data['detail']['detectiveUrls']['privateIpAddress']  = prefix_url+data['region']+'#entities/IpAddress/'+v[index]['privateIpAddress']+'?scopeStart='+scopeStart+'&scopeEnd='+scopeEnd
+                    else:
+                        data['detail']['detectiveUrls']['privateIpAddress'+str(index + 1)]  = prefix_url+data['region']+'#entities/IpAddress/'+v[index]['privateIpAddress']+'?scopeStart='+scopeStart+'&scopeEnd='+scopeEnd
+                    index += 1
+            
+            ## Get public IP addresses from network interfaces
+            publicIPs = find_key_value_pairs(data['detail'], 'publicIp')
+            
+            index = 0
+            if len(publicIPs) > 0:
+                for k, v in publicIPs:
+                    if len(publicIPs) == 1:
+                        data['detail']['detectiveUrls']['publicIpAddress']  = prefix_url+data['region']+'#entities/IpAddress/'+v+'?scopeStart='+scopeStart+'&scopeEnd='+scopeEnd
+                    else:
+                        data['detail']['detectiveUrls']['publicIpAddress'+str(index + 1)]  = prefix_url+data['region']+'#entities/IpAddress/'+v+'?scopeStart='+scopeStart+'&scopeEnd='+scopeEnd
+                    index += 1
+ 
+        ## Get External IP addresses from services
+        ## Get External local ip addresses
+        localIpAddress = find_key_value_pairs(data['detail'], 'localIpDetails')
+        
+        index = 0
+        if len(localIpAddress) > 0:
+            for k, v in localIpAddress:
+                if len(localIpAddress) == 1:
+                    data['detail']['detectiveUrls']['localIpAddress']  = prefix_url+data['region']+'#entities/IpAddress/'+v['ipAddressV4']+'?scopeStart='+scopeStart+'&scopeEnd='+scopeEnd
+                else:
+                    data['detail']['detectiveUrls']['localIpAddress'+str(index + 1)]  = prefix_url+data['region']+'#entities/IpAddress/'+v['ipAddressV4']+'?scopeStart='+scopeStart+'&scopeEnd='+scopeEnd
+                index += 1
+
+        return_event['sourcetype'] = st
+        return_event['event'] = data['detail']
+
+        ## Get External Remote ip addresses
+        remoteIpAddress = find_key_value_pairs(data['detail'], 'remoteIpDetails')
```

Comment:
Move below line:210 because you're appending fields and values like remoteIpDetails in the `data` map.